ccmlib/scylla_repository.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import with_statement
+
+import os
+import tarfile
+import tempfile
+import time
+import shutil
+import glob
+
+from six import print_
+from six.moves import urllib
+
+from ccmlib.common import (ArgumentError, CCMError, get_default_path, rmdirs, validate_install_dir)
+from ccmlib.repository import __download
+
+GIT_REPO = "http://github.com/scylladb/scylla.git"
+
+RELOCATABLE_URLS_BASE = 'https://s3.amazonaws.com/downloads.scylladb.com/relocatable/{0}/{1}'
+
+
+def setup(version, verbose=True):
+    s3_url = 'https://s3.amazonaws.com/downloads.scylladb.com/relocatable/unstable/master/latest'
+    type_n_version = version.split(':')
+    if len(type_n_version) == 2:
+        s3_version = type_n_version[1]
+        s3_url = RELOCATABLE_URLS_BASE.format(type_n_version[0], s3_version)
+        version = os.path.join(*type_n_version)
+
+    cdir = version_directory(version)
+    if cdir is None:
+        url = os.environ.get('SCYLLA_PACKAGE', os.path.join(s3_url, 'scylla-package.tar.gz'))
+        download_version(version, verbose=verbose, url=url)
+
+        url = os.environ.get('SCYLLA_JAVA_TOOLS_PACKAGE', os.path.join(s3_url, 'scylla-tools-package.tar.gz'))
+        download_version(os.path.join(version, 'scylla-java-tools'), verbose=verbose, url=url)
+
+        # hack to make the relocatable tools work
+        cdir = version_directory(version)
+        scylla_java_tools_dir = os.path.join(cdir, 'scylla-java-tools')
+        for jar_file in glob.glob(scylla_java_tools_dir + "/*.jar"):
+            shutil.copy(jar_file, os.path.join(scylla_java_tools_dir, "lib"))
+
+        url = os.environ.get('SCYLLA_JMX_PACKAGE', os.path.join(s3_url, 'scylla-jmx-package.tar.gz'))
+        download_version(os.path.join(version, 'jmx'), verbose=verbose, url=url)
+
+        cdir = version_directory(version)
+    return cdir, version
+
+
+min_attributes = ('scheme', 'netloc')
+
+
+def is_valid(url, qualifying=None):
+    qualifying = min_attributes if qualifying is None else qualifying
+    token = urllib.parse.urlparse(url)
+    return all([getattr(token, qualifying_attr)
+                for qualifying_attr in qualifying])
+
+
+def download_version(version, url=None, verbose=False):
+    """Download, extract, and build Cassandra tarball.
```

Comment:
Cassandra tarball ?

Reply:
copy pasted from Cassandra CCM , I'll rewrite this comment